mirrord-layer/src/socket/ops.rs

```diff
@@ -210,167 +275,36 @@ pub(super) fn connect(sockfd: RawFd, remote_address: SocketAddr) -> HookResult<(
-            // Just call `libc::connect`.
-            trace!(
-                "connect -> SocketState::Initialized {:#?}",
-                user_socket_info
-            );
-
-            let rawish_remote_address = SockAddr::from(remote_address);
-            let result = unsafe {
-                FN_CONNECT(
-                    sockfd,
-                    rawish_remote_address.as_ptr(),
-                    rawish_remote_address.len(),
-                )
-            };
-
-            if result != 0 {
-                let io_error = io::Error::last_os_error();
-
-                match io_error.kind() {
-                    io::ErrorKind::Uncategorized => io_error
-                        .raw_os_error()
-                        .map(|blocking_error| {
-                            if blocking_error == libc::EINPROGRESS || blocking_error == libc::EINTR
-                            {
-                                Ok(())
-                            } else {
-                                Err(io_error)?
-                            }
-                        })
-                        .unwrap(),
-                    _ => Err(io_error)?,
-                }
-            } else {
-                Ok::<_, HookError>(())
-            }
+    match user_socket_info.kind {
+        SocketKind::Udp(_) if enabled_udp_outgoing => {
+            connect_outgoing::<UDP>(sockfd, remote_address, user_socket_info)
         }
-        SocketState::Initialized if !((is_ignored_port(port)) && (is_ignored_ip(ip))) => {
-            // Prepare this socket to be intercepted.
-            trace!(
-                "connect -> SocketState::Initialized {:#?}",
-                user_socket_info
-            );
-            let (mirror_tx, mirror_rx) = oneshot::channel();
-
-            let connect = Connect {
-                remote_address,
-                channel_tx: mirror_tx,
-            };
-
-            let connect_hook = TcpOutgoing::Connect(connect);
-
-            blocking_send_hook_message(HookMessage::TcpOutgoing(connect_hook))?;
-            let MirrorAddress(mirror_address) = mirror_rx.blocking_recv()??;
-
-            let connect_to = SockAddr::from(mirror_address);
-
-            // Connect to the interceptor socket that is listening.
-            let connect_result =
-                unsafe { FN_CONNECT(sockfd, connect_to.as_ptr(), connect_to.len()) };
-
-            let err_code = errno::errno().0;
-            if connect_result == -1 && err_code != libc::EINPROGRESS && err_code != libc::EINTR {
-                error!(
-                    "connect -> Failed call to libc::connect with {:#?} errno is {:#?}",
-                    connect_result,
-                    errno::errno()
-                );
-                return Err(io::Error::last_os_error())?;
+        SocketKind::Tcp(_) => match user_socket_info.state {
+            SocketState::Initialized if !((is_ignored_port(port)) && (is_ignored_ip(ip))) => {
```

Comment:
So we use `is_ignored_port` together with `is_ignored_ip` here, but on its own above, which seems weird? Can you explain it?

Reply:
This came in #370. I'll remove the `is_ignored_port` duplicate.